[API] Boolean CF not correctly set on PostgreSQL

NobodysNightmare opened the pull request:
## OpenProject work package

https://community.openproject.org/work_packages/19601

NobodysNightmare committed:
add integrative tests for typed_value for booleans

first charge of tests.

I intend those to fail for postgres, but can't verify that locally, so travis
will have to do it for me.

If you see this build failing for the newly added specs: This is intentional!

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://nobodysnightmare-feature-db-agnostic-booleans.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://nobodysnightmare-feature-db-agnostic-booleans.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

NobodysNightmare committed:
make transform to boolean more database resistant

PostgreSQL has many representations for bool: http://www.postgresql.org/docs/9.1/static/datatype-boolean.html

NobodysNightmare committed:
add roundtrip tests for more datatypes

NobodysNightmare commented:
Note that I have intentionally left two travis tasks running on https://github.com/NobodysNightmare/openproject/commit/27d6abfa35dbc57bdaa8110578b4f91768df47f3, because I want to verify that the specs effectively find the flaw.

NobodysNightmare commented:
Not sure whether this really fixes the issues we had on QA systems... Specs don't reproduce the issue.

Probably merge + retest on QA?
